### DISCUSSION TITLE
BRICSBuild not reconstructing original molecule..

### DISCUSSION THREAD

**Discussion by DanielGoede** [rdkit/rdkit#5297] [General]
I am trying to learn more about the BRICS implementation in rdkit, but running into a peculiarity.
I can fragment a molecule easily, but when I try to re-assemble it, the enumerated molecules
either (a) don't use all of the available fragments, or (b) use the same fragments over and over.

For example:

```python
from rdkit import Chem
from rdkit.Chem import BRICS
from rdkit import DataStructs


>> mol = Chem.MolToSmiles('Cc1ncc(-c2cc(Nc3ncnc4c(NC(N)=O)cccc34)cc(C(F)F)c2)[nH]1')
>> fragments = BRICS.BRICSDecompose(mol)
>> print(fragments)

{'[14*]c1ncnc2c([16*])cccc12', '[14*]c1cnc(C)[nH]1', '[1*]C(N)=O', '[5*]N[5*]', '[16*]c1cc([16*])cc([16*])c1', '[8*]C(F)F'}

>> fragments = [Chem.MolFromSmiles(x) for x in fragments]
>> combined_mols = BRICS.BRICSBuild(fragments)
>> print([Chem.MolToSmiles(x) for x in combined_mols][0:3])
>>
Cc1ncc(-c2cccc3c(-c4cccc5c(-c6cccc7c(C(F)F)ncnc67)ncnc45)ncnc23)[nH]1
c1ncc(Nc2ncnc3c(Nc4cnc(C)[nH]4)cccc23)[nH]1
Cc1ncc(Nc2ncnc3c(C(F)F)cccc23)[nH]1
...

I was under the impression that the original molecule would be in the list, but it doesn't appear to be:

```python
>>> base_fps = Chem.RDKFingerprint(m)
>>> combined_fps = [Chem.RDKFingerprints(x) for x in combined_mols]
>>> similarities = [DataStructs.FingerprintSimilarity(base_fps, x) for x in combined_fps]
>>> max(similarities)
0.6258445945945946
```

Is there a way to ensure that the original molecule is reconstructed in this process? I would like to be
able to manipulate this process for core hopping by changing the new scaffolds to match the fragmentation labels
on the old core and reconstruct as close as possible, but can't seem to make it work...

**Comment by greglandrum** [comment]
Hi @DanielGoede sorry I'm so slow to reply to this.

`BRICSBuild()` has a `maxDepth` argument which controls how many fragments can be combined to create the final molecules. This defaults to 3 in order to control the size of the output (which can get very big very quickly)

Since your molecule produces 6 fragments, you would have to go all the way to a maxDepth of 5 in order to find the input molecule. This will take quite a while and generate a *bunch* of molecules (I didn't let it run to completion, so I'm not sure how many).

If you want to check that the fragmentation does in fact return the original molecule, you need to either start with a smaller input structure or be prepared to wait. :-)
